selfservice/strategy/password/settings_test.go

```diff
@@ -158,20 +157,30 @@ func TestSettings(t *testing.T) {
 		})
 
 		t.Run("session=needs reauthentication", func(t *testing.T) {
-			conf.MustSet(ctx, config.ViperKeySelfServiceSettingsPrivilegedAuthenticationAfter, "1ns")
 			defer testhelpers.NewLoginUIWith401Response(t, conf)
-			t.Cleanup(func() {
-				conf.MustSet(ctx, config.ViperKeySelfServiceSettingsPrivilegedAuthenticationAfter, "5m")
-			})
+
+			var sessionMutator func(mutateFunc func(session *session.Session))
+			browserUser1 := testhelpers.NewHTTPClientBuilder(t).
+				SetReqestFromWhoAmI().
+				SetIdentity(browserIdentity1).
+				SetSessionWithProvider(testhelpers.UnprivilegedProvider()).
+				RecordSessionMutator(reg, &sessionMutator).
```

Comment:
Here we're able to get a pointer to a function (saved to `sessionMutator`) for mutating the session at a later point. It's used on like 201 to later set the session to privileged, as if the user had refreshed their session.